integration/helpers.go

```diff
@@ -1547,15 +1547,76 @@ func startAndWait(process *service.TeleportProcess, expectedEvents []string) ([]
 	return receivedEvents, nil
 }
 
-type proxyServer struct {
+type proxyBasicAuthHandler struct {
```

Comment:
nit: Can we rename this to something like:   basicAuthMiddleware to emphasise that it wraps the proxyHandler ?